src/controller/CHIPDevice.h

```diff
@@ -473,6 +475,7 @@ class DLL_EXPORT Device : public Messaging::ExchangeDelegate, public SessionEsta
     CASESession mCASESession;
 
     Credentials::OperationalCredentialSet * mCredentials = nullptr;
+    uint8_t * mCredentialsIndex                          = nullptr;
```

Comment:
If it's an index, why is it a uint8_t pointer?

Reply:
Because all other members of the ControllerDeviceInitParams struct are pointers. I did it like this to follow the convention.
Also, I think it makes sense to rely on the true credentialsIndex variable which is present in DeviceController class

Reply:
It is entirely possible to use the Device object outside of the `chip::Device::DeviceController`.  Or stated another way, it's possible (and currently necessary in some cases) to roll your own controller.  Dependencies like this on the `chip::Device::DeviceController` object are going to create problems.

I think this also shows a larger problem with the `OperationalCredentialSet` object, which can only hold `kOperationalCredentialsMax` credentials.  Why?  This is only the spec-candidate minimum.  If I want to hold 1000 sets of credentials and I can fit these in memory, I should be able to.

I think we need an object that holds only a single set of credentials and that's what this  object, the channel, should reference.  It's impossible for a channel to use multiple credentials simultaneously, so why does it need a handle to more than one?

Reply:
I agree with this, but I think it should be addressed in a separate PR because the OperationalCredentialSet requires a complete refactor.

Reply:
Could you please open an issue related to the follow-up and add its number to the TODOs you had to address later?

Reply:
Will do this.